src/cli/create.py

```diff
@@ -1,247 +1,339 @@
+                raise exceptions.InvalidSubcharacteristic(
+                    "{} does not have weight value inside parameters (0 to 100).".format(
+                        subcharacteristic["name"]
+                    )
+                )
+
+            if "weight" in subcharacteristic.keys():
+                sum_of_subcharacteristics_weights = (
+                    sum_of_subcharacteristics_weights + subcharacteristic["weight"]
+                )
+
+            if "measures" not in subcharacteristic.keys():
+                raise exceptions.InvalidSubcharacteristic(
+                    "{} does not have measures field defined.".format(
+                        subcharacteristic["name"]
+                    )
+                )
+
+            if (
+                subcharacteristic["measures"] is None
+                or len(subcharacteristic["measures"]) == 0
+            ):
+                raise exceptions.InvalidSubcharacteristic(
+                    "{} needs to have at least one measure defined.".format(
+                        subcharacteristic["name"]
+                    )
+                )
+
+            sub_characteristics_names.append(subcharacteristic["name"])
+
+        sum_of_subcharacteristics_weights = round_sum_of_weights(
+            sum_of_subcharacteristics_weights
         )
 
-    def validation_func(values):
-        return validate_check_box_input(len(values))
+        if validate_sum_of_weights(sum_of_subcharacteristics_weights) is False:
+            raise exceptions.InvalidSubcharacteristic(
+                "The sum of subcharacteristics weights is not 100"
+            )
 
-    return generic_valid_input(
-        validation_func,
-        body_func,
-        VALID_CHECKBOX_ERROR,
-    )
+    return True
 
 
-def generic_valid_input(
-    validation_function, body_function, error_message, initial_value=None
-):
-    values = [] if initial_value is None else initial_value
-    valid_input = validation_function(values)
+def validate_file_measures(preconfig_json_file):
 
-    while not valid_input:
-        values = body_function()
+    measures_names = []
 
-        valid_input = validation_function(values)
+    for characteristic in preconfig_json_file["characteristics"]:
+        for subcharacteristic in characteristic["subcharacteristics"]:
 
-        print_error_message(error_message, valid_input)
+            sum_of_measures_weights = 0
 
-    return values
+            for measure in subcharacteristic["measures"]:
 
+                if "name" not in measure.keys():
+                    raise exceptions.InvalidMeasure("Expected measure name field.")
 
-def input_weights(sublevels_config, user_sublevels):
-    def body_func():
-        sublevels_weights = []
-        for x in user_sublevels:
-            sublevels_weights.append(define_weight(x, sublevels_config[x]["name"]))
-        return sublevels_weights
+                if "weight" not in measure.keys():
```

Comment:
É melhor fazer esse `measure.keys()` e salvar numa variável e usar nos locais. Chamar isso várias vezes tem o custo de pegar as chaves (deve ser bem otimizado, mas mesmo assim, não é legal.

Outro ponto para vocês olharem é que tem muita coisa parecida aqui, principalmente essa parte de `if not in` e não sei o que. A lógica é sempre a mesma dá para criar uma função e só passar parâmetros para customizar.